components/ActionForm.tsx

```diff
@@ -0,0 +1,150 @@
+import SelectUnit from "./select_unit";
+import SelectResourceType from "./select_resource_type";
+import React, {useState} from "react";
+import {useAuth} from "../lib/auth";
+import {DocumentNode, TypedDocumentNode, useMutation} from "@apollo/client";
+import SelectUser from "./SelectUser";
+import SelectInventoriedResource from "./SelectInventoriedResource";
+
+type ActionVariables = {
+    inputOf?:string,
+    provider:string,
+    receiver: string,
+    resourceInventoriedAs?:string,
+    outputOf?: string,
+    newInventoriedResource?: { name: string, note: string },
+    resourceConformsTo?: string,
+    resourceQuantity: { hasNumericalValue: number, hasUnit: string },
+    hasPointInTime: string,
+}
```

Comment:
🥰 For not putting the types in a separate files, unless are not shared between pages